packages/docs/docs/advanced/custom-components.mdx

```diff
@@ -0,0 +1,141 @@
+---
+slug: /custom-components
+---
+
+import AnimationPlayer from '@site/src/components/AnimationPlayer';
+import CodeBlock from '@theme/CodeBlock';
+import source from '!!raw-loader!@motion-canvas/examples/src/scenes/components-switch';
+
+# Components
+
+<AnimationPlayer name="components" banner />
+
+Components are reusable parts of code such as `Rect` and `Circle` which can be
+ scenes. To use a component in a scene, add it to the view and
+provide arguments to the component.
+
+```tsx
+<Switch
+    initialState={false}
+/>
+```
+To define what arguments a component will take in, first define an interface. All 
+properties of the interface must be wrapped in [SignalValue<>](/api/core/signals#SignalValue) 
+as such:
+```tsx
+//  You can extend an existing props interface
+//  such as LayoutProps, ShapeProps or NodeProps to
+//  include their properties alongside the ones you 
+//  define
+
+export interface SwitchProps extends NodeProps {
+    initialState?: SignalValue<boolean>
+
+    // We dont use color here because we want
+    // to be able to pass hex strings and rgb
+    // values to accent rather than a `Color`
+    accent?: SignalValue<PossibleColor>
+}
+```
+Next, create a component class for your components. The component class must extend [Node](/api/2d/components/Node)
+or one of its subclasses. If you dont want to inherit any methods from an existing component, extend your class from
+`Node`. It is advisable to extend from a component most similar to the custom component you are building. For instance, 
```

Comment:
We should probably keep with the plural first person here: "We advise extending from ~~a~~ the component most similar to the ~~custom~~ component you are building"

Reply:
done!